nf_core/schema.py

```diff
@@ -163,56 +166,42 @@ def build_schema(self, pipeline_dir, no_prompts, web_only, url):
         if url:
             self.web_schema_build_url = url
 
-        # Load a JSON Schema file if we find one
-        self.schema_filename = os.path.join(pipeline_dir, 'nextflow_schema.json')
-        if(os.path.exists(self.schema_filename)):
-            logging.debug("Parsing existing JSON Schema: {}".format(self.schema_filename))
-            try:
-                self.load_schema()
-            except Exception as e:
-                logging.error("Existing JSON Schema found, but it is invalid:\n {}".format(click.style(str(e), fg='red')))
-                logging.info(
-                    "Please fix or delete this file, then try again.\n" \
-                    "For more details, run the following command:\n  " + \
-                    click.style("nf-core schema lint {}".format(self.schema_filename), fg='blue')
-                )
-                sys.exit(1)
-            logging.info("Loaded existing JSON schema with {} params: {}\n".format(len(self.schema['properties']), self.schema_filename))
-        else:
-            logging.debug("Existing JSON Schema not found: {}".format(self.schema_filename))
-
-        # Build a skeleton schema if none already existed
-        if not self.schema:
+        # Get JSON Schema filename
+        try:
+            self.get_schema_from_name(pipeline_dir, local_only=True)
+        except AssertionError:
             logging.info("No existing schema found - creating a new one from scratch")
-            self.schema_from_scratch = True
-            config = nf_core.utils.fetch_wf_config(pipeline_dir)
-            self.schema = {
-                "$schema": "http://json-schema.org/draft-07/schema",
-                "$id": "https://raw.githubusercontent.com/{}/master/nextflow_schema.json".format(config['manifest.name']),
-                "title": "{} pipeline parameters".format(config['manifest.name']),
-                "description": config['manifest.description'],
-                "type": "object",
-                "properties": {}
-            }
+            self.make_skeleton_schema()
+
+        # Load and validate Schema
+        try:
+            self.lint_schema()
+        except AssertionError as e:
+            logging.error("Existing JSON Schema found, but it is invalid: {}".format(click.style(str(self.schema_filename), fg='red')))
+            logging.info("Please fix or delete this file, then try again.")
```

Comment:
I don't know how good the schema linting is, but could we return fields which should be fixed in this message?

Reply:
Yup, the linting already logs error messages itself so these will be printed above this message.